packages/camera/lib/new/src/support_android/support_android_configurator.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../common/camera_interface.dart';
+import '../common/native_texture.dart';
+import 'camera.dart';
+import 'camera_info.dart';
+
+/// Default [CameraConfigurator] for [CameraApi.supportAndroid].
+///
+/// This is used as the default [CameraConfigurator] for Android sdks below 21
+/// when using [CameraController].
+///
+/// This can also be used independently of [CameraController] when one needs
+/// greater control of a camera on Android sdks below 21.
+class SupportAndroidConfigurator implements CameraConfigurator {
+  SupportAndroidConfigurator(this.info) : assert(info != null);
+
+  NativeTexture _texture;
+  Camera _camera;
+  bool _isDisposed = false;
+
+  final CameraInfo info;
+
+  Camera get camera => _camera;
+
+  @override
+  Future<int> addPreviewTexture() {
```

Comment:
So just making sure I understand this: it creates a preview texture if one doesn't exist, but if one does it exist it always returns the existing one?

I think that behavior is reasonable but it's not what I would have guessed from the method name. `getOrCreatePreviewTexture` maybe? Would be good for the docs on the interface to mention the behavior too.

Reply:
Added to interface documentation here: https://github.com/flutter/plugins/pull/1930/commits/08ef2931ac4b9bb269ff4f1c5c0bba3a9273ed70